src/containers/edit-playlist/mobile/EditPlaylistPage.tsx

```diff
@@ -179,15 +199,30 @@ const EditPlaylistPage = g(
       if (formFields.description === undefined) {
         formFields.description = null
       }
+      // Copy the metadata playlist contents so that a reference is not changed between
+      // removing tracks, updating track order, and edit playlist
+      const playlistTrackIds = [
+        ...(metadata?.playlist_contents?.track_ids ?? [])
+      ]
+
+      for (const removedTrack of removedTracks) {
+        const { playlist_id } = metadata!
+        removeTrack(removedTrack.trackId, playlist_id, removedTrack.timestamp)
```

Comment:
this remove flow needs to use UID. remove doesn't work if multiple tracks of same id in the playlist i think

Reply:
I think the timestamp differentiates the tracks, so it should be fine.

Reply:
lmk if it this doesn't work. I just pushed a fix for unique draggable track list item which might have been the bug you saw.